### PR TITLE
CryptoPkg/BaseCryptLib: add additional RSAES-OAEP crypto functions

### PR DESCRIPTION
**Pull request by cmruffin**
## Description

Expand the availability of the RSAES-OAEP crypto capability in
BaseCryptLib. Applications using RSA crypto functions directly from
OpensslLib can transition to BaseCryptLib to take advantage of the
shared crypto feature in CryptoDxe.

- [ ] Impacts functionality?
- [x] Impacts security?
- [ ] Breaking change?
- [x] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

Host-based unit tests, end-to-end testing with shared crypto binary.

## Integration Instructions

Depends on similar PR in mu_basecore

### CODE REVIEW

File: CryptoBinPkg/Driver/Bin/Crypto.pcd.MINIMAL_SHA_SM3.inc.dsc
Comment (makubacki) [inline CryptoBinPkg/Driver/Bin/Crypto.pcd.MINIMAL_SHA_SM3.inc.dsc:1]: I know we haven't followed this until very recently, but we've found history can be cleaned up quite a bit by removing files with only timestamp changes from check-in. Can you please drop them here?

File: CryptoBinPkg/Driver/Bin/Crypto.inc.dsc
Comment (makubacki) [inline CryptoBinPkg/Driver/Bin/Crypto.inc.dsc:1]: @cmruffin did you observe a noticeable size increase with this change?

Comment (cmruffin) [reply CryptoBinPkg/Driver/Bin/Crypto.inc.dsc:1]: CryptoDxe increased by 3072 bytes.

**Review by makubacki — APPROVED** [review]

**Review by apop5 — APPROVED** [review]

**Comment by cmruffin** [comment]
update MU_BASECORE rev

**Review by kenlautner — APPROVED** [review]

**Comment by cmruffin** [comment]
@makubacki this is ready to go pointing with pointing at mu_basecore:personal/v-chruffin/rsaoaep with headers in place

**Comment by makubacki** [comment]
> @makubacki this is ready to go pointing at mu_basecore:personal/v-chruffin/rsaoaep with headers in place

The `MU_BASECORE` submodule in this repo is currently pointing to the `release/202302` branch (this repo was not moved to `release/202311` due to OpenSSL 3 and its issues). We've been okay with this because of minimal dependencies in these crypto binaries on Mu Basecore. However, for consistency, I'd like to kick the build off with your Mu Basecore changes on top of `release/202302`.

- I pushed that branch here (the commit in the submodule needs to reside in the microsoft repo for CI): https://github.com/microsoft/mu_basecore/commits/personal/cmruffin/rsaoaep_202302/
- To keep a branch in this repo that is relatively stable and can be used to find the source of the NuGet binary build, I pushed this branch based on your changes but pointing to that mu_basecore branch: https://github.com/microsoft/mu_crypto_release/commits/mucryptobin_2023_2_14/
- To make it clear this binary was not generated from release branches, I added the `-rc` label as a postfix to the version, so it will be `2023.2.14-rc`. This means a build of the same content from the mu_crypto_release `release/202302` branch can be made after this PR completes as `2023.2.14` which will have the expected provenance on the release branches.

**Comment by makubacki** [comment]
> > @makubacki this is ready to go pointing at mu_basecore:personal/v-chruffin/rsaoaep with headers in place
> 
> The `MU_BASECORE` submodule in this repo is currently pointing to the `release/202302` branch (this repo was not moved to `release/202311` due to OpenSSL 3 and its issues). We've been okay with this because of minimal dependencies in these crypto binaries on Mu Basecore. However, for consistency, I'd like to kick the build off with your Mu Basecore changes on top of `release/202302`.
> 
> * I pushed that branch here (the commit in the submodule needs to reside in the microsoft repo for CI): https://github.com/microsoft/mu_basecore/commits/personal/cmruffin/rsaoaep_202302/
> * To keep a branch in this repo that is relatively stable and can be used to find the source of the NuGet binary build, I pushed this branch based on your changes but pointing to that mu_basecore branch: https://github.com/microsoft/mu_crypto_release/commits/mucryptobin_2023_2_14/
> * To make it clear this binary was not generated from release branches, I added the `-rc` label as a postfix to the version, so it will be `2023.2.14-rc`. This means a build of the same content from the mu_crypto_release `release/202302` branch can be made after this PR completes as `2023.2.14` which will have the expected provenance on the release branches.

The binary is now published as `2023.2.14-rc`. You should be able to use that as the version in the ext-dep over in mu_basecore.

**Comment by makubacki** [comment]
This PR needs to be updated to target the newly pushed [`release/202311`](https://github.com/microsoft/mu_crypto_release) branch.

**Comment by makubacki** [comment]
> This PR needs to be updated to target the newly pushed [`release/202311`](https://github.com/microsoft/mu_crypto_release) branch.

I went ahead and pushed that PR here - https://github.com/microsoft/mu_crypto_release/pull/74. Closing this PR.

**Pull request closed without merging**